helix-term/src/commands.rs

```diff
@@ -1169,6 +1171,26 @@ fn search_impl(doc: &mut Document, view: &mut View, contents: &str, regex: &Rege
     };
 }
 
+fn search_completions(cx: &mut Context, reg: Option<char>) -> Vec<String> {
+    let mut exist_elems = std::collections::BTreeSet::new();
+    reg.and_then(|reg| cx.editor.registers.get(reg))
+        .map_or(Vec::new(), |reg| {
+            reg.read()
+                .iter()
+                .filter(|&item| {
+                    if exist_elems.contains(item) {
+                        false
+                    } else {
+                        exist_elems.insert(item);
+                        true
+                    }
+                })
+                .take(10)
+                .cloned()
+                .collect()
+        })
+}
```

Comment:
I don't get why you need to `take(10)` or iterate through it when `cx.editor.registers.get(reg)` only return an option. It's just one item at most.